test-rt/suite-unit/src/q_binary.rs

```diff
@@ -0,0 +1,397 @@
+                        .to_array_view()
+                        .unwrap()
+                        .iter()
+                        .any(|x: &f32| *x == 0.0))
+            })
+            .boxed()
+    }
+}
+
+impl Test for QBinaryOpProblem {
+    fn run_with_approx(
+        &self,
+        _suite: &str,
+        id: &str,
+        runtime: &dyn Runtime,
+        approx: Approximation,
+    ) -> infra::TestResult {
+        let mut model = TypedModel::default();
+        model.properties.insert("tract-rt-test.id".to_string(), rctensor0(id.to_string()));
+
+        let a = model.add_source(
+            "a",
+            TypedFact::dt_shape(self.tensor_a.datum_type(), self.tensor_a.shape()),
+        )?;
+        let b = model.add_const("b", self.tensor_b.clone().into_arc_tensor())?;
+        // we need to wire correctly output to the provided operator {
+        let mut op = self.operator.clone();
+        op.1 = Some(self.c_dt);
+        // }
+        let c = model.wire_node("c", op, &[a, b])?[0];
+        model.set_output_outlets(&[c])?;
+
+        let result = runtime
+            .prepare(model)?
+            .run(tvec![self.tensor_a.clone().into_tvalue()])?
+            .remove(0)
+            .into_tensor();
+        let mut reference = self.reference_float_ops()?;
+
+        let (zero_point, scale) = self.c_dt.zp_scale();
+        let min_repr_val = (self.c_dt.unquantized().min_value().cast_to_scalar::<f32>()?
+            - zero_point as f32)
+            * scale;
+        let max_repr_val = (self.c_dt.unquantized().max_value().cast_to_scalar::<f32>()?
+            - zero_point as f32)
+            * scale;
+
+        reference
+            .to_array_view_mut()?
+            .iter_mut()
+            .for_each(|x: &mut f32| *x = (*x).clamp(min_repr_val, max_repr_val));
+
+        let mut diff = result.cast_to::<f32>()?.into_owned();
+
+        let acceptable_scale_error_ratio = match approx {
+            Approximation::Exact => 0.,
+            Approximation::Approximate => 1.,
+            _ => 2.,
+        };
+        tract_core::ndarray::Zip::from(diff.to_array_view_mut()?)
+            .and(reference.to_array_view()?)
+            .all(|x: &mut f32, xref: &f32| {
+                let closest_x = (*x).clamp(min_repr_val, max_repr_val);
+                // core maximal accepted distance by default
+                (xref - closest_x).abs() <= scale * acceptable_scale_error_ratio
+            });
+        Ok(())
+    }
+}
+
+pub fn suite() -> TractResult<TestSuite> {
+    let mut suite = TestSuite::default();
+    suite.add_arbitrary::<QBinaryOpProblem>("proptest", ());
+
+    // simplification 0 at declutter constant
+    suite.add(
+        "trivial_mul_0_case",
+        QBinaryOpProblem {
+            operator: tract_core::ops::math::mul(),
+            tensor_a: tensor0(0u8)
+                .cast_to_dt(
+                    u8::datum_type().quantize(QParams::ZpScale { zero_point: 0, scale: 1. }),
```

Comment:
I think if we define something like

```
fn qu8(zp: i32, scale: f32) -> DatumType { ... } 
```

we could reduce widely the bloat in test definition. WDYT ?

Note that `?` should actually work there instead of `unwrap()`.